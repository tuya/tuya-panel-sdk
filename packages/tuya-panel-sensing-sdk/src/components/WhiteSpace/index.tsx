import React, { FC } from 'react';
import { StyleProp, View, ViewStyle } from 'react-native';

export interface WhiteSpaceProps {
  style?: StyleProp<ViewStyle>;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | number;
}

const SPACE_INFO = {
  v_spacing_xs: 2,
  v_spacing_sm: 4,
  v_spacing_md: 12,
  v_spacing_lg: 16,
  v_spacing_xl: 20,
};

const WhiteSpace: FC<WhiteSpaceProps> = props => {
  const { size, style } = props;
  return (
    <View
      style={[
        { height: typeof size === 'number' ? size : SPACE_INFO[`v_spacing_${size!}`] },
        style,
      ]}
    />
  );
};

WhiteSpace.defaultProps = {
  size: 'sm',
};

export default WhiteSpace;
