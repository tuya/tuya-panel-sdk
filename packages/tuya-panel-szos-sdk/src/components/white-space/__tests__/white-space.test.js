/*
 * @Author: 豆芽(douya.ye@tuya.com)
 * @Date: 2021-11-16 14:38:23
 * @LastEditTime: 2021-11-16 14:43:40
 * @LastEditors: 豆芽(douya.ye@tuya.com)
 * @Description: 上下留白
 * @FilePath: /tuya-panel-sdk/packages/tuya-panel-szos-sdk/src/components/white-space/__tests__/white-space.test.js
 */

import React from 'react';
import { View } from 'react-native';
import { shallow, mount } from 'enzyme';
import WhiteSpace from '../index';

describe('BoxShow components', () => {
  it('default render', () => {
    const wrapper = shallow(<WhiteSpace />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render height', () => {
    const wrapper = shallow(<WhiteSpace height={10} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render backgroundColor', () => {
    const wrapper = shallow(<WhiteSpace backgroundColor="green" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render all', () => {
    const wrapper = shallow(<WhiteSpace backgroundColor="green" height={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});
