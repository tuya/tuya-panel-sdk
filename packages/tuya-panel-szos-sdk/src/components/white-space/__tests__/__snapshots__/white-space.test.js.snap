// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxShow components default render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 4,
    }
  }
/>
`;

exports[`BoxShow components render all 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "height": 10,
    }
  }
/>
`;

exports[`BoxShow components render backgroundColor 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "height": 4,
    }
  }
/>
`;

exports[`BoxShow components render height 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 10,
    }
  }
/>
`;
